### DISCUSSION TITLE
Shouldn't the hovercard be shown on focus too?

### DISCUSSION THREAD

**Discussion by gnapse** [ariakit/ariakit#2237] [Questions] [answered]
The hovercard appears on mouse over, but shouldn't it appear when the anchor element gets focus too? For reference, that's how [radix-ui's hovercard](https://www.radix-ui.com/docs/primitives/components/hover-card) component works.

**Comment by diegohaz** [comment, marked as answer]
I don't think so.

Hovercards often have interactive controls inside. Look at the Facebook hovercard (that also appears on focus), for example:

<img width="461" alt="Screenshot of the Facebook hovercard with interactive controls inside" src="https://user-images.githubusercontent.com/3068563/231178953-15c36236-fd7e-40d9-ba27-5b4bdd0a27fb.png">

Question: as a sighted keyboard user, how do I interact with the contents inside the hovercard?

Pressing <kbd>Tab</kbd> will move focus to the element next to the hovercard anchor, which automatically hides the card. Even if <kbd>Tab</kbd> moved the focus to the hovercard, it could be really annoying, adding many additional tab-stops to the page. Maybe there's a hidden shortcut to access the hovercard contents. But if it's not obvious, it doesn't matter.

As for the Radix example, I'm not sure if it's a bug or a feature, as it mentions `Opens on hover only` in the feature list.

If the hovercard should be perceivable by keyboard and screen reader users, I recommend checking the [Hovercard with keyboard support](https://ariakit.org/examples/hovercard-disclosure) example.

---

That all said, if you really want to show the hovercard on focus, you can do something like this:

```jsx
<div>
  <HovercardAnchor ref={anchorRef} onFocusVisible={hovercard.show} />
  <Hovercard hideOnHoverOutside={() => document.activeElement !== anchorRef.current} />
</div>
```

**Reply by andreoperiboschi** [reply]
Additionally, this mechanism is compliant with WCAG criterion 1.4.13 https://www.w3.org/WAI/WCAG21/quickref/#content-on-hover-or-focus

**Comment by gnapse** [comment]
Indeed, after asking this, I realized that while radix-ui's hovercard appears on focus in practice, [their documentation says that it is intended for mouse users only](https://www.radix-ui.com/docs/primitives/components/hover-card#accessibility).

In a way, it makes sense. I guess the key here is to understand that hovercards are a convenience interaction, and that, ultimately, keyboard-centric users should have ways to achieve whatever the hovercard offers, but through different ways (most likely clicking the anchor, of course).

Thanks for your input. We can close this now.
